src/Dsll.hs

```diff
@@ -0,0 +1,397 @@
+-- | "DSLL" contains the grammers and parser for the 
+--    DSLL language
+--    Author: Dor Ma'ayan, May 2018
+
+{-# OPTIONS_HADDOCK prune #-}
+--module DSLL where
+module Main (main) where -- for debugging purposes
+
+import System.Process
+import Control.Monad (void)
+import Data.Void
+import Debug.Trace
+import System.IO -- read/write to file
+import System.Environment
+import Control.Arrow ((>>>))
+import System.Random
+import Debug.Trace
+import Data.Functor.Classes
+import Data.List
+import Data.Maybe (fromMaybe)
+import Data.Typeable
+import Text.Megaparsec
+import Text.Megaparsec.Char
+import Text.Megaparsec.Expr
+--import Text.PrettyPrint
+--import Text.PrettyPrint.HughesPJClass hiding (colon, comma, parens, braces)
+import qualified Data.Map.Strict as M
+import qualified Text.Megaparsec.Char.Lexer as L
+--------------------------------------- DSLL AST ---------------------------------------
+type TypeVar = String
```

Comment:
I would still keep the constructors here even though there's only one branch in the ADT:
`data TypeVar = TVar String`, and analogously for Var
The reasoning is that we'll probably be doing a lot of debugging by examining the pretty-printed AST, and it's useful to see what kind of var or name something is, rather than have everything be a string
(I guess you could also put this information in the pretty-printer rather than the ADT)